### DISCUSSION TITLE
Hasura data provider with custom primary keys

### DISCUSSION THREAD

**Discussion by mKoder** [refinedev/refine#4515] [Q&A] [answered]
Hey, we are using Hasura but we don't use the name "id" as the primary key. It looks like the provided has "id" hard coded in a few places. Can we use the out-of-the-box Hasura data provider if we have non-standard primary keys please?

Any tips / advice?

Thank you!

**Comment by BatuhanW** [comment, marked as answer]
Hello @mKoder, you can swizzle the data provider and modify t as you need. 

https://refine.dev/docs/tutorial/understanding-dataprovider/swizzle/#how-do-you-use-swizzle-to-create-an-data-provider

**Reply by mKoder** [reply]
Thank you, i'll check that out
